lib/node/nodes/kmeans.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'kmeans';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    clusters : Node.PARAM.NUMBER
+};
+
+var KMeans  = Node.create(TYPE, PARAMS, {cache: false});
```

Comment:
We will need to `{cache: true}` this analysis as it need to be run at whole dataset level, otherwise it will run at per [meta]tile level so it won't make sense at all.

`cache: true`
![screen shot 2016-06-09 at 13 30 17](https://cloud.githubusercontent.com/assets/45346/15928319/8231a018-2e46-11e6-8400-a5b86cd5af53.png)

`cache: false`
![screen shot 2016-06-09 at 13 30 54](https://cloud.githubusercontent.com/assets/45346/15928331/8eca3a24-2e46-11e6-8569-b214f612e605.png)


cc @javisantana